api/http/handlerfuncs_backup_test.go

```diff
@@ -0,0 +1,743 @@
+	require.Equal(t, http.StatusBadRequest, errResponse.Errors[0].Extensions.Status)
+	require.Equal(t, "Bad Request", errResponse.Errors[0].Extensions.HTTPError)
+	require.Equal(t, "only JSON format is supported at the moment", errResponse.Errors[0].Message)
+}
+
+func TestExportHandler_WithInvalidCollection_ReturnError(t *testing.T) {
+	t.Cleanup(CleanupEnv)
+	env = "dev"
+	ctx := context.Background()
+	defra := testNewInMemoryDB(t, ctx)
+	defer defra.Close(ctx)
+
+	filepath := t.TempDir() + "/test.json"
+	cfg := client.BackupConfig{
+		Filepath:    filepath,
+		Format:      "json",
+		Collections: []string{"invalid"},
+	}
+	b, err := json.Marshal(cfg)
+	require.NoError(t, err)
+	buf := bytes.NewBuffer(b)
+
+	errResponse := ErrorResponse{}
+	testRequest(testOptions{
+		Testing:        t,
+		DB:             defra,
+		Method:         "POST",
+		Path:           ExportPath,
+		Body:           buf,
+		ExpectedStatus: 400,
+		ResponseData:   &errResponse,
+	})
+	require.Contains(t, errResponse.Errors[0].Extensions.Stack, "collection does not exist: datastore: key not found")
+	require.Equal(t, http.StatusBadRequest, errResponse.Errors[0].Extensions.Status)
+	require.Equal(t, "Bad Request", errResponse.Errors[0].Extensions.HTTPError)
+	require.Equal(t, "collection does not exist: datastore: key not found", errResponse.Errors[0].Message)
+}
+
+func TestExportHandler_WithBasicExportError_ReturnError(t *testing.T) {
+	t.Cleanup(CleanupEnv)
+	env = "dev"
+	db := mocks.NewDB(t)
+	testError := errors.New("test error")
+	db.EXPECT().BasicExport(mock.Anything, mock.Anything).Return(testError)
+
+	filepath := t.TempDir() + "/test.json"
+	cfg := client.BackupConfig{
+		Filepath: filepath,
+	}
+	b, err := json.Marshal(cfg)
+	require.NoError(t, err)
+	buf := bytes.NewBuffer(b)
+
+	errResponse := ErrorResponse{}
+	testRequest(testOptions{
+		Testing:        t,
+		DB:             db,
+		Method:         "POST",
+		Path:           ExportPath,
+		Body:           buf,
+		ExpectedStatus: 500,
+		ResponseData:   &errResponse,
+	})
+	require.Contains(t, errResponse.Errors[0].Extensions.Stack, "test error")
+	require.Equal(t, http.StatusInternalServerError, errResponse.Errors[0].Extensions.Status)
+	require.Equal(t, "Internal Server Error", errResponse.Errors[0].Extensions.HTTPError)
+	require.Equal(t, "test error", errResponse.Errors[0].Message)
+}
+
+// func TestExportHandler_WithGetCollectionError_ReturnError(t *testing.T) {
+// 	t.Cleanup(CleanupEnv)
+// 	env = "dev"
+// 	db := mocks.NewDB(t)
+// 	testError := errors.New("test error")
+// 	db.EXPECT().GetAllCollections(mock.Anything).Return(nil, testError)
+
+// 	errResponse := ErrorResponse{}
+// 	testRequest(testOptions{
+// 		Testing:        t,
+// 		DB:             db,
+// 		Method:         "GET",
+// 		Path:           ExportPath,
+// 		Body:           nil,
+// 		ExpectedStatus: 500,
+// 		ResponseData:   &errResponse,
+// 	})
```

Comment:
todo: uncomment?

Reply:
They need to be removed. I had kept them as reference for tests I moved to the `db` package. Thanks for pointing this out.